[18.0][IMP] edi_exchange_template_oca: add json support

wouitmil opened the pull request:
This PR aims to merge the module edi_exchange_template_oca_json in edi_exchange_template_oca

If a template is of type json. payload key is waited in the result. result['payload'] contains then the generated json

simahawk reviewed (changes requested):
This change should be done in the base module.

Side note: do not add the odoo version in the commit https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#commit-message

wouitmil commented:
> This change should be done in the base module.
> 
> Side note: do not add the odoo version in the commit https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#commit-message

In fact, shouldn’t all these modules be merged into the base module? JSON is quite common, and it doesn’t require any dependencies.

Commit:
[IMP] edi_exchange_template: add validator for code snippet

Commit:
[IMP] edi_exchange_template: add json generator

[IMP] edi_exchange_template: added _evaluate_code_snippet_validate_json and patched test to comply with it

Commit:
[REM] edi_exchange_template_oca_json

This feature has been merged into edi_exchange_template_oca.

simahawk committed:
[IMP] edi_exchange_template: unify postprocessor lookup

Rely on specific method as per validator and code executor.

simahawk commented:
@wouitmil I've cleanup commits and changed the lookup for the post processor.

simahawk commented:
/ocabot merge minor
